### ISSUE TITLE
SendMoneyToDestination FAILED: ...This transaction requires...

### ISSUE THREAD

**Issue by osnwt** [eXtremal-ik7/pool_frontend_zcash#19]
Built the pool for testnet (changed base58 prefixes). Seems all to be ok except payment. Here is a log:

SendMoneyToDestination FAILED: 5.8714 coins to tmGBVGKA8qJyVz38AK1BwAmkEvnYApTgNuF because: Error: This transaction requires a transaction fee of at least 0.00 because of its amount, complexity, or use of recently received funds!

More logs:

<info> accounting: balance=0.00 req/balance=52.8666 req/queue=52.8666 immature=64.6245 users=52.8666 queued=11.7568, net=0.0011
<info> clients: 1, workers: 1, cpus: 0, gpus: 0, asics: 0, other: 0, latency: 0, power: 24533
 new block signal: 11765, diff=3688.94171, approximate shares per block: 461
<info> Accounting: checking 14 payout requests...
<info> Accounting: [0] SendMoneyToDestination FAILED: 5.8714 coins to tmGBVGKA8qJyVz38AK1BwAmkEvnYApTgNuF because: Error: This transaction requires a transaction fee of at least 0.00 because of its amount, complexity, or use of recently received funds!
<info> clients: 1, workers: 1, cpus: 0, gpus: 0, asics: 0, other: 0, latency: 0, power: 24533

What could be the reason? What are requirements for successful transaction?

**Comment by nginnever** [comment]
Your balance is zero so there is no amount to cover the tx fee. I assume you mined a block and are waiting for it to mature.

**Comment by osnwt** [comment]
Yes, I mined 11 blocks, but have never had deals with full nodes. So have
no idea how long a mined block matures to get paid for it.

**Comment by nginnever** [comment]
Zcash requires 100 blocks. Give it time and this issue will go away.

**Comment by osnwt** [comment]
Oh, thank you, got it!

**Comment by osnwt** [comment]
Yes, it seems to proceed. I will close the issue. But last question: am I right that funds go like this:

- coinbase-generated coins for the block found go to new t-addr
- from this t-addr they go to pool z-addr
- from pool z-addr they go to pool t-addr
- from pool t-addr they go to client t-addr and pool fee t-addr

**Comment by nginnever** [comment]
From what I have observed yes.

**Comment by osnwt** [comment]
An interesting thing was observed: playing with the pool, I tried to mine using extra t-addrs generated on the same pool/zcashd assuming there will be no interference between pool addrs and these addrs. Suddenly, I found that coins are being transferred in a loop between t-addrs and z-addrs. It seems that in such configuration pool makes infinite never ending transfers.

Of course, this is not a use case it was written for. Though, it may be good to notice that such config leads to loops (is it?), and how to stop it if occured.
